### PR TITLE
support for bitcoin parsing and formatting

### LINKED ISSUE #97: Bitcoin not supported by stringToUnits()
Because stringToUnits() explicitly expects only 2 trailing decimals, it cannot support Bitcoin and we see "The value could not be parsed as money"

### PR DESCRIPTION
**Pull request by frederikbosch**
Fixes #97.

### CODE REVIEW

**Comment by frederikbosch** [comment]
I just found out that StyleCI automatically pushes a PR for style inconsistencies. Since I removed the inconsistencies by hand already, I removed those PRs.

File: src/Formatter/BitcoinSupportedMoneyFormatter.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Money\Formatter;
+
+use Money\Money;
+use Money\MoneyFormatter;
+use Money\Parser\BitcoinSupportedMoneyParser;
+
+class BitcoinSupportedMoneyFormatter implements MoneyFormatter
```
Comment (sagikazarmark) [inline src/Formatter/BitcoinSupportedMoneyFormatter.php]: Author is missing

File: src/Formatter/BitcoinSupportedMoneyFormatter.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+     */
+    private $innerFormatter;
+    /**
+     * @var int
+     */
+    private $fractionDigits;
+
+    /**
+     * @param MoneyFormatter $innerFormatter
+     * @param $fractionDigits
```
Comment (sagikazarmark) [inline src/Formatter/BitcoinSupportedMoneyFormatter.php]: Type is missing

File: src/Formatter/BitcoinSupportedMoneyFormatter.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+use Money\Parser\BitcoinSupportedMoneyParser;
+
+class BitcoinSupportedMoneyFormatter implements MoneyFormatter
+{
+    const CODE = 'XBT';
+
+    /**
+     * @var MoneyFormatter
+     */
+    private $innerFormatter;
```
Comment (sagikazarmark) [inline src/Formatter/BitcoinSupportedMoneyFormatter.php]: Let's call this delegatedFormatter

File: src/Formatter/BitcoinSupportedMoneyFormatter.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        $negative = false;
+
+        if (substr($valueBase, 0, 1) === '-') {
+            $negative = true;
+            $valueBase = substr($valueBase, 1);
+        }
+
+        $fractionDigits = $this->fractionDigits;
+        $valueLength = strlen($valueBase);
+        if ($valueLength > $fractionDigits) {
```
Comment (sagikazarmark) [inline src/Formatter/BitcoinSupportedMoneyFormatter.php]: Add some spacing before ifs to increase readability

Comment (sagikazarmark) [inline src/Formatter/BitcoinSupportedMoneyFormatter.php]: Before control structures in general

File: src/Parser/BitcoinSupportedMoneyParser.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Money\Parser;
+
+use Money\Currency;
+use Money\Formatter\BitcoinSupportedMoneyFormatter;
+use Money\Money;
+use Money\MoneyParser;
+
+class BitcoinSupportedMoneyParser implements MoneyParser
```
Comment (sagikazarmark) [inline src/Parser/BitcoinSupportedMoneyParser.php]: Author is missing

File: src/Parser/BitcoinSupportedMoneyParser.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+{
+    const SYMBOL = "\0xC9\0x83";
+
+    /**
+     * @var MoneyParser
+     */
+    private $innerParser;
```
Comment (sagikazarmark) [inline src/Parser/BitcoinSupportedMoneyParser.php]: Let's call this delegated as well.

File: src/Parser/BitcoinSupportedMoneyParser.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * @var int
+     */
+    private $fractionDigits;
+
+    /**
+     * @param MoneyParser $innerParser
+     * @param $fractionDigits
```
Comment (sagikazarmark) [inline src/Parser/BitcoinSupportedMoneyParser.php]: Type is missing

File: src/Formatter/BitcoinSupportedMoneyFormatter.php (hunk 1 above)
Comment (sagikazarmark) [inline src/Formatter/BitcoinSupportedMoneyFormatter.php]: I can't see a reason why it couldn't be final. WDYT?

**Comment by sagikazarmark** [comment]
LGTM. I added a few styling comments.

**Comment by frederikbosch** [comment]
What is wrong with Travis? Why does it not start?

**Comment by sagikazarmark** [comment]
no clue, travis DERP. Restarted.

File: src/Formatter/BitcoinSupportedMoneyFormatter.php [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+ */
+final class BitcoinSupportedMoneyFormatter implements MoneyFormatter
+{
+    const CODE = 'XBT';
+
+    /**
+     * @var MoneyFormatter
+     */
+    private $delegatedFormatter;
+    /**
```
Comment (sagikazarmark) [inline src/Formatter/BitcoinSupportedMoneyFormatter.php]: Missing new line

File: src/Parser/BitcoinSupportedMoneyParser.php [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+ */
+final class BitcoinSupportedMoneyParser implements MoneyParser
+{
+    const SYMBOL = "\0xC9\0x83";
+
+    /**
+     * @var MoneyParser
+     */
+    private $delegatedParser;
+    /**
```
Comment (sagikazarmark) [inline src/Parser/BitcoinSupportedMoneyParser.php]: Missing newline

**Comment by sagikazarmark** [comment]
Two comments added.

This is fine from my point of view. I don't use Bitcoin, if anyone has problem with it, we can treat it as bugfix.

**Comment by frederikbosch** [comment]
Ok. will add the new lines, squash and then merge.

**Comment by sagikazarmark** [comment]
:+1:

**Pull request merged**